ananse/influence.py

```diff
@@ -456,6 +457,120 @@ def save_reg_network(self, filename, full_output=False):
                         d["TF_act_target"],
                     ]
                     nw.write("\t".join(str(v) for v in cols) + "\n")
+#def plot_TF_GRN(self, infile, outfile, GRN_wb = False, network_algorithm = 'neato', n_TFs = 20, cmap = 'viridis', full_output = False):
+    def plot_TF_GRN(self, infile, outfile, GRN_wb = False, network_algorithm = 'neato', n_TFs = 20, cmap = 'viridis', full_output = False, cutoff_val = 0.1):
+        """
+        Plot the top20 of differential TFs and their interactions with the highest interaction score into a GRN network image
+
+        Parameters
+        ----------
+        self: influence_object to extract the diff network self.G, #future optimization = read diffnetwork instead
+        infile: influence output file
+        outfile: output file location
+        GRN_wb: when true and a -full--output influence GRN is suplied, use weighted binding interaction score to vizualize
+                edges instead of the probability score
+        network_algorithm: pyviz cluster algorithm used for node placement, options include: neato, dot, fdp, twopi, sfdp, circo 
+        n_TFs: number of (significantly differential expressed) Tfs to vizualize
+        cmap: matplotlib colour pallet used
+        full_output: if the GRN is a full output GRN (usefull for ploting weighted binding instead of prob score)
+        cutoff_val: minimum value the prob/wb score needs to have to be included in the GRN, 0.1 seems like a good number.
+    """
+        
+        cmap=plt.get_cmap(cmap)
+        mogrify = pd.read_table(infile, index_col="factor")
+        mogrify = mogrify[mogrify.GscoreScaled > 0] #plot only TFs that are differential 
+        factors = list(mogrify.sort_values("sumScaled").tail(n_TFs).index)#get the top TFs to include in the GRN    
+        #get the mean interaction score of the selected top TFs
+        def filter_node(n1):
+            return n1 in  factors
+        TF_G = nx.subgraph_view(self.G, filter_node=filter_node)#filter the diffnetwork to only contain topTF-topTF
```

Comment:
`self.G` is the only part that it needs from `class Influence`.

could you reconstruct `G` from a file (output of `save_reg_network()`)? If so we can move this function to a separate command (both in python and the command line)...

Reply:
See my earlier 3 points (bellow the plot examples), this is one of them :)

Reply:
Should be fixed in the newest PR update :)